community/kernel/src/test/java/org/neo4j/kernel/api/index/NonUniqueIndexAccessorCompatibility.java

```diff
@@ -173,6 +176,20 @@ public void testIndexRangeSeekByPrefixWithDuplicates() throws Exception
     }
 
     @Test
+    public void testIndexRangeSeekByPrefixWithDuplicatesWithIndexQuery() throws Exception
+    {
+        updateAndCommit( asList(
+                IndexEntryUpdate.add( 1L, index, "a" ),
+                IndexEntryUpdate.add( 2L, index, "A" ),
+                IndexEntryUpdate.add( 3L, index, "apa" ),
+                IndexEntryUpdate.add( 4L, index, "apa" ),
+                IndexEntryUpdate.add( 5L, index, "apa" ) ) );
+
+        assertThat( query( stringPrefix( 1, "a" ) ), equalTo( asList( 1L, 3L, 4L, 5L ) ) );
+        assertThat( query( stringPrefix( 1, "apa" ) ), equalTo( asList( 3L, 4L, 5L ) ) );
+    }
+
+    @Test
```

Comment:
Will we remove these older tests and the older API once the new stuff is well tested?